### DISCUSSION TITLE
Adding "fromSecretNumbers" to the Wallet class

### DISCUSSION THREAD

**Discussion by nixer89** [XRPLF/xrpl.js#1788] [Ideas]
I am getting a lot of questions from new developers which want to build on the XRPL. Mostly they start with having a XUMM wallet and now want to develop something. The most common question is: "How to get the 'secret key' from my XUMM wallet?"

While the XUMM wallet uses [SecretNumbers](https://www.npmjs.com/package/xrpl-secret-numbers), it does not allow to export the secret key / family seed.

I think it would be nice if the Wallet instance of xrpl.js would offer a "Wallet.fromSecretNumbers" method to make it easier for new developers to start building.

**Comment by nixer89** [comment]
a possible and easy solution could be to import:

`import {
  Utils
} from 'xrpl-secret-numbers'`

And implement it like this:

![image](https://user-images.githubusercontent.com/25486793/141090763-52a8d77d-40b9-47a2-8ea1-f6e54fba3f39.png)

**Reply by JST5000** [reply]
That seems reasonable to me, would you be willing to put up a PR for that?

**Comment by nixer89** [comment]
yes sure, I will do that tomorrow! :-)

**Reply by JST5000** [reply]
Sweet, looking forward to reviewing it :)

**Comment by justinr1234** [comment]
Associated PR: https://github.com/XRPLF/xrpl.js/pull/1799

**Comment by JST5000** [comment]
Took a while, but finally got this in for the next release of xrpl.js :)
